upb/upb.h

```diff
@@ -196,10 +196,17 @@ template <int N> class InlinedEnvironment;
 
 #define UPB_UNUSED(var) (void)var
 
-/* For asserting something about a variable when the variable is not used for
- * anything else.  This prevents "unused variable" warnings when compiling in
- * debug mode. */
-#define UPB_ASSERT_VAR(var, predicate) UPB_UNUSED(var); assert(predicate)
+/* UPB_ASSERT(): in release mode, we use the expression without letting it be
+ * evaluated.  This prevents "unused variable" warnings. */
+#ifdef NDEBUG
+#define UPB_ASSERT(expr) if (false && (expr)) {}
```

Comment:
Nit: I think this macro could theoretically cause problems if you use it inside an if statement without braces. E.g. I think the code below would do something unexpected:

```
if (whatever)
  UPB_ASSERT(foo)
else
  // ...
```

In this case I think the else statement will always be executed because it will associate itself with the if statement inside the macro. Maybe this would be one way to fix it:

`#define UPB_ASSERT(expr) do {} while (false && (expr))`

If I'm not mistaken this also has the advantage of requiring a semicolon at the end when the macro is used, so then there's no ambiguity about whether you need to put a semicolon or not.